pkg/athena/api/api.go

```diff
@@ -2,34 +2,41 @@ package api
 
 import (
 	"context"
+	"errors"
+	"fmt"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
 	"github.com/aws/aws-sdk-go/service/athena"
 	"github.com/aws/aws-sdk-go/service/athena/athenaiface"
 	"github.com/grafana/athena-datasource/pkg/athena/models"
 	"github.com/grafana/grafana-aws-sdk/pkg/awsds"
+	"github.com/grafana/grafana-aws-sdk/pkg/sql/api"
+	"github.com/grafana/sqlds/v2"
 )
 
 type API struct {
 	Client   athenaiface.AthenaAPI
 	settings *models.AthenaDataSourceSettings
 }
 
-func New(sessionCache *awsds.SessionCache, settings *models.AthenaDataSourceSettings) (*API, error) {
-	region := settings.DefaultRegion
-	if settings.Region != "" {
-		region = settings.Region
-	}
-	session, err := sessionCache.GetSession(region, settings.AWSDatasourceSettings)
+func New(sessionCache *awsds.SessionCache, settings *models.AthenaDataSourceSettings) (api.AWSAPI, error) {
+	sess, err := awsds.GetSessionWithDefaultRegion(sessionCache, settings.AWSDatasourceSettings)
 	if err != nil {
 		return nil, err
 	}
-	return &API{athena.New(session), settings}, nil
+
+	svc := athena.New(sess)
+	svc.Handlers.Send.PushFront(func(r *request.Request) {
```

Comment:
https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#hdr-Adding_Handlers
Seems like you can attach a handler directly to the session. In that case, we can push the user agent code to the GetSessionWithDefaultRegion func in the sdk.

Reply:
good find! I am not sure about modifying `GetSessionWithDefaultRegion` because that would be a breaking change (adding a new parameter), even though only redshift and athena seems to use that. We can add a new method for that. I am going to open a new PR for that and then I will update this.